src/Roslyn.Diagnostics.Analyzers/Core/AbstractApplyTraitToClass.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Roslyn.Diagnostics.Analyzers
+{
+    public abstract class AbstractApplyTraitToClass<TAttributeSyntax> : CodeRefactoringProvider
+        where TAttributeSyntax : SyntaxNode
+    {
+        protected AbstractApplyTraitToClass()
+        {
+        }
```

Comment:
Can be deleted I think. The ctor of abstract classes is protected by default.

Reply:
⏱ Will remove